core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -51,8 +56,9 @@ protected void onPrepareFailure(Snapshot snapshot) {}
   protected void onValidateFailure(Snapshot snapshot) {}
 
   public void commit(Snapshot snapshot) throws CommitException, UnknownTransactionStatusException {
+    Optional<PreparedSnapshotHookFuture> snapshotHookFuture;
     try {
-      prepare(snapshot);
+      snapshotHookFuture = prepare(snapshot);
```

Comment:
Maybe we can move the snapshotHandler logic outside the `prepare()` method as follows?
```suggestion
      snapshotHandleFuture = handleSnapshotBeforePreparation(snapshot);
      prepare(snapshot);
```

What do you think?

Reply:
Ah, good point! I missed the improvement after we decided to use Snapshot itself... I'll update the PR.

Reply:
Addressed them in https://github.com/scalar-labs/scalardb/pull/2271/commits/067d69df83b55412f8c8147f71882d892da2a258.

@brfrn169  I also noticed the original PR didn't properly handle errors when invoking the snapshot hook and waiting the snapshot future, resulting in throwing non-CommitException and resource leaks. I enhanced the error handling in the commit. PTAL!